drivers/arm/i2c_defs.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#if defined(STM32F1XX) || defined(STM32F1xx) || defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32L0xx) || defined(STM32L1xx)
+    #define USE_I2CV1
+#endif
+
+#ifdef I2C1_BANK
+    #define I2C1_SCL_BANK I2C1_BANK
+    #define I2C1_SDA_BANK I2C1_BANK
+#endif
+
+#ifndef I2C1_SCL_BANK
+    #define I2C1_SCL_BANK GPIOB
+#endif
+
+#ifndef I2C1_SDA_BANK
+    #define I2C1_SDA_BANK GPIOB
+#endif
+
+#ifndef I2C1_SCL
+    #define I2C1_SCL 6
+#endif
+#ifndef I2C1_SDA
+    #define I2C1_SDA 7
+#endif
+
+#ifdef USE_I2CV1
+    #ifndef I2C1_OPMODE
+        #define I2C1_OPMODE OPMODE_I2C
+    #endif
+    #ifndef I2C1_CLOCK_SPEED
+        #define I2C1_CLOCK_SPEED 100000 /* 400000 */
+    #endif
+    #ifndef I2C1_DUTY_CYCLE
+        #define I2C1_DUTY_CYCLE STD_DUTY_CYCLE /* FAST_DUTY_CYCLE_2 */
+    #endif
+#else
+    // The default PAL alternate modes are used to signal that the pins are used for I2C
+    #ifndef I2C1_SCL_PAL_MODE
+        #define I2C1_SCL_PAL_MODE 4
+    #endif
+    #ifndef I2C1_SDA_PAL_MODE
+        #define I2C1_SDA_PAL_MODE 4
+    #endif
+
+    // The default timing values below configures the I2C clock to 400khz assuming a 72Mhz clock
```

Comment:
this comment may be incorrect now based on configuring 100khz

Reply:
OK, I see how I mis-read this. Do we want to go with 400 for i2cv2? I think bakingpy reported issues with slower MCUs or poor quality cables.

Reply:
I think that will need some testing, to be honest. 

But I think for now, 100khz may be the safest solution

Reply:
Ideally, the behaviour should be matched, and ARM I2C should use 400khz, and drop down to 100khz when running split via `quantum/split_common/post_config.h`. When running on a GPIOv1 board, that's easy but GPIOv2 is an issue. The timer configuration is not a single value and hardware dependent.

Now I have a real board, I can at least test the 100/400khz for GPIOv1.